src/journal-remote/journal-remote.c

```diff
@@ -777,21 +788,37 @@ static int setup_microhttpd_socket(RemoteServer *s,
         return setup_microhttpd_server(s, fd, key, cert, trust);
 }
 
+static int null_timerfd_handler(sd_event_source *event, uint64_t usec, void *userdata)
+{
+    return dispatch_http_event(event, 0, 0, userdata);
+}
+
 static int dispatch_http_event(sd_event_source *event,
                                int fd,
                                uint32_t revents,
                                void *userdata) {
         MHDDaemonWrapper *d = userdata;
         int r;
+        fd_set rd, w, e;
+        unsigned long long timeout = -1ULL;
 
         assert(d);
 
-        r = MHD_run(d->daemon);
+        MHD_get_fdset (d->daemon, &rd, &w, &e, 0);
+
+        r = MHD_run_from_select(d->daemon, &rd, &w, &e);
         if (r == MHD_NO) {
                 log_error("MHD_run failed!");
                 // XXX: unregister daemon
                 return -EINVAL;
         }
+        if (MHD_get_timeout(d->daemon, &timeout) == MHD_NO) {
+            sd_event_source_set_time(d->timerfd, -1ULL);
+        }
```

Comment:
please follow coding style: indent by multiples of 8ch, and no {} around single line if blocks. Also please -1ULL is kinda ugly (since it is self-contradictory: you specify a signed value but then declare it to be unsigned, which C is fine with, but is still ugly. Also unsigned long long is not the right type here anyway, this is a uint64_t, hence use UINT64_MAX or so)